components/provisioner/internal/operations/stages/provisioning/connect_agent.go

```diff
@@ -34,11 +39,15 @@ func (s *ConnectAgentStep) TimeLimit() time.Duration {
 
 func (s *ConnectAgentStep) Run(cluster model.Cluster, _ model.Operation, _ logrus.FieldLogger) (operations.StageResult, error) {
 
-	if cluster.Kubeconfig == nil {
-		return operations.StageResult{}, fmt.Errorf("error: kubeconfig is nil")
+	var kubeconfig []byte
+	{
+		var err error
+		kubeconfig, err = s.dynamicKubeconfigProvider.FetchFromRequest(cluster.ClusterConfig.Name)
+		if err != nil {
```

Comment:
This is a single chance to get kubeconfig to setup runtime agent. So instead of failing the whole step  it is beeter to schedule a retry here.

 ```suggestion
		if err != nil {
		     return operations.StageResult{Stage: s.Name(), Delay: 20 * time.Second}, nil
```

Reply:
Fixed by myself